components/popup/RateDish.vue

```diff
@@ -84,14 +86,20 @@ const inputNickname = useLocalStorage('rate-dish--nickname', () => '')
 
 const errorRequireStars = useState(`rate-dish-${props.occurrence.id}--error-stars`, () => false)
 const submittedLoading = useState(`rate-dish-${props.occurrence.id}--submitted-loading`, () => false)
-const readyToSubmit = computed(() => !!inputStars.value)
+const readyToSubmit = computed(() => !!inputStars.value && currentImageId.value !== 'uploading')
 
 watch(inputStars, () => errorRequireStars.value = false)
 
+const currentImageId = useState<string | null>(`rate-dish-${props.occurrence.id}--current-image-id`, () => null)
+
 const { open, onChange } = useFileDialog()
-onChange((files) => {
-  if (files?.length)
+onChange(async (files) => {
+  if (files?.length) {
     inputImage.value = files[0]
```

Comment:
why are you keeping inputImage here, if it's no longer used anywhere? I'd rename currentImageId to inputImageId to be more descriptive and remove inputImage completely then.

Reply:
Isn't it required for `fileUploadPreview`?